src/utils/cognito.ts

```diff
@@ -0,0 +1,17 @@
+import { CognitoIdentityServiceProvider } from 'aws-sdk';
+import { MAIN_USER_POOL_PREFIX } from '../constants';
+import { getCognitoClient, getUserPoolList } from '../data/cognito';
+
+interface CognitoClientAndUserPool {
+    client: CognitoIdentityServiceProvider;
+    userPoolId: string | undefined;
+}
+
+const getCognitoClientAndUserPool = async (): Promise<CognitoClientAndUserPool> => {
+    const client = await getCognitoClient();
+    const userPoolList = await getUserPoolList(client);
+    const userPool = userPoolList?.find((pool) => pool.Name?.startsWith(MAIN_USER_POOL_PREFIX));
+    return { client, userPoolId: userPool?.Id };
```

Comment:
It looks like we're doing `if (userPoolId)` checks when we use this method elsewhere, because it might return undefined. Instead, could we throw the error here if its not found, and then we know it will always return what we want.